tests/test_contracts.py

```diff
@@ -101,6 +103,29 @@ def parse_no_url(self, response):
         pass
 
 
+class TestSameUrlSpider(Spider):
+
+    name = 'test_same_url'
+
+    def __init__(self, *args, **kwargs):
+        super(TestSameUrlSpider, self).__init__(*args, **kwargs)
+        self.visited = 0
+
+    def parse_first(self, response):
+        """first callback
+        @url http://scrapy.org
```

Comment:
Is it going to be a real http request to scrapy.org? If so, this would make tests flaky. We have a mockserver available in tests, so it is better to use it - https://github.com/scrapy/scrapy/blob/91f986ecf46e3618b14eac6aa4d46bf57e20e266/tests/test_crawl.py#L16 is an usage example.